tests/test_openclhpp.cpp

```diff
@@ -3378,4 +3415,395 @@ void testCommandBufferInfoKHRCommandQueues()
 #endif
 }
 
+/****************************************************************************
+ * Tests for cl::Semaphore
+ ****************************************************************************/
+#if defined(cl_khr_semaphore)
+void testMoveAssignSemaphoreNonNull(void);
+void testMoveAssignSemaphoreNull(void);
+void testMoveConstructSemaphoreNonNull(void);
+void testMoveConstructSemaphoreNull(void);
+MAKE_MOVE_TESTS(Semaphore, make_semaphore_khr, clReleaseSemaphoreKHR, semaphorePool);
+#else
+void testMoveAssignSemaphoreNonNull(void) {}
+void testMoveAssignSemaphoreNull(void) {}
+void testMoveConstructSemaphoreNonNull(void) {}
+void testMoveConstructSemaphoreNull(void) {}
+#endif
+
+static cl_int clEnqueueWaitSemaphoresKHR_testEnqueueWaitSemaphores(
+    cl_command_queue command_queue,
+    cl_uint num_sema_objects,
+    const cl_semaphore_khr* sema_objects,
+    const cl_semaphore_payload_khr* sema_payload_list,
```

Comment:
It'd be best to put all of these new tests inside of one big `#if defined(cl_khr_semaphore)`.

If the headers don't support semaphore (and hence `cl_khr_semaphore` is not defined) then `cl_semaphore_khr` and other types that these tests use won't be defined, either.

Reply:
Done.